engine/runtime-compiler/src/main/scala/org/enso/compiler/Compiler.scala

```diff
@@ -575,7 +575,6 @@ class Compiler(
             u => {
               u.resetScope()
               u.ir(updatedIr)
-              u.compilationStage(CompilationStage.AFTER_PARSING)
```

Comment:
I would prefer: `u.compilationStage(CompilationStage.AFTER_STATIC_PASSES)`. If not, then at least add `assert module.getCompilationStage() == CompilationStage.AFTER_STATIC_PASSES` to see if the assumption is always correct.